src/libstore/build/local-derivation-goal.cc

```diff
@@ -645,12 +646,32 @@ void LocalDerivationGoal::startBuilder()
                 dirsInChroot.erase(worker.store.printStorePath(*i.second.second));
         }
 
-#elif __APPLE__
-        /* We don't really have any parent prep work to do (yet?)
-           All work happens in the child, instead. */
+        if (useSystemdCgroup) {
+            settings.requireExperimentalFeature("systemd-cgroup");
```

Comment:
Is there anything that speaks against just using cgroup namespaces and create a new hierarchy?
Than this would not depend on what cgroups are available outside and leak less of information/build impurities.